ScanbotSDKExamples/MAUI/ClassicComponent.Maui/Platforms/iOS/CustomViews/BarcodeCameraView_iOS.cs

```diff
@@ -0,0 +1,152 @@
+                SetConfigurations();
+            }
+        }
+
+        private async Task<UIViewController> CheckViewControllerAvailable(BarcodeCameraViewHandler barcodeCameraViewHandler)
+        {
+            var retryCount = 5;
+
+            UIKit.UIViewController viewController = null;
+            do
+            {
+                await Task.Delay(500);
+                viewController = AppDelegate.ExtractViewController(barcodeCameraViewHandler?.PlatformView?.Window);
+                var message = viewController != null ? "available" : "unavailable";
+                System.Diagnostics.Debug.WriteLine("ViewController is " + message);
+                retryCount++;
+            } while (viewController == null || retryCount < 5);
+            return viewController;
+        }
+
+        #region Properties Implementation
+
+        internal void MapIsFlashEnabled(bool isFlashEnabled)
+        {
+            if (cameraViewController == null) return;
+            cameraViewController.FlashLightEnabled = isFlashEnabled;
+        }
+
+        // -----------------------------------------
+        // Selection Overlay Config binding
+        // -----------------------------------------
+        internal void MapOverlayConfiguration(BarcodeCameraView commonView)
+        {
+            if (cameraViewController == null) return;
+            var config = commonView.OverlayConfiguration;
+            if (config.Enabled == true)
+            {
+                cameraViewController.SelectionOverlayEnabled = true;
+
+                cameraViewController.SelectionPolygonColor = config.PolygonColor.ToNative();
+                cameraViewController.SelectionTextColor = config.TextColor.ToNative();
+                cameraViewController.SelectionTextContainerColor = config.TextContainerColor.ToNative();
+                if (config.HighlightedPolygonColor != null)
+                {
+                    cameraViewController.SelectionHighlightedPolygonColor = config.HighlightedPolygonColor?.ToNative();
+                }
+
+                if (config.HighlightedTextColor != null)
+                {
+                    cameraViewController.SelectionHighlightedTextColor = config.HighlightedTextColor?.ToNative();
+                }
+
+                if (config.HighlightedTextContainerColor != null)
+                {
+                    cameraViewController.SelectionHighlightedTextContainerColor = config.HighlightedTextContainerColor?.ToNative();
+                }
+            }
+        }
+
+        #endregion
+
+        #region Event Handlers Implementation
+
+        internal void MapStartDetectionHandler()
+        {
+            if (cameraViewController == null) return;
+            cameraViewController.RecognitionEnabled = true;
+        }
+
+        internal void MapStopDetectionHandler()
+        {
+            if (cameraViewController == null) return;
+            cameraViewController.RecognitionEnabled = false;
+        }
```

Comment:
Why are these methods called "MapXYZ" couldn't they just be: "StartDetectionHandler" or "SetFlashEnabled" and so forth?

Reply:
Yes, as discussed on call, it is a mapping so we will keep it, or we can also rename it to Handle. Let me know.